tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,606 @@
+					scNamePrefix,
+				)
+				scName = scObj.ObjectMeta.Name
+
+				By("building a PVC with StorageClass " + scName)
+				pvcObj, err = BuildPersistentVolumeClaim(pvcName, scName, capacity, accessModes)
+				Expect(err).ShouldNot(
+					HaveOccurred(),
+					"while building PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("creating above PVC with StorageClass " + scName)
+				createdPvc, err := ops.PVCClient.WithNamespace(namespaceObj.Name).Create(context.TODO(), pvcObj)
+				Expect(err).To(
+					BeNil(),
+					"while creating PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("building a pod with busybox image")
+				podObj, err = BuildPod(podName, pvcName, labelselector)
+				Expect(err).ShouldNot(
+					HaveOccurred(),
+					"while building pod {%s} in namespace {%s}",
+					podName,
+					namespaceObj.Name,
+				)
+
+				By("creating above pod with busybox image")
+				createdPod, err := ops.PodClient.WithNamespace(namespaceObj.Name).
+					Create(context.TODO(), podObj)
+				Expect(err).To(
+					BeNil(),
+					"while creating pod {%s} in namespace {%s}",
+					podName,
+					namespaceObj.Name,
+				)
+
+				By("verifying pod status as not running")
+				podPhase := ops.GetPodStatusEventually(createdPod)
+				Expect(podPhase).ToNot(Equal(corev1.PodRunning), "while verifying pod running status")
+
+				By("verifying the above created pvc phase")
+				Expect(createdPvc.Status.Phase).To(Equal(corev1.ClaimPending), "while verifying the pvc state")
+			})
+		})
+
+		When("Pod consuming pvc with a valid quota storageclass is deleted along with pvc and storageclass", func() {
+			It("should delete pod, pvc and storageclass", func() {
+				By("deleting above pod")
+				err = ops.PodClient.WithNamespace(namespaceObj.Name).Delete(context.TODO(), podName, &metav1.DeleteOptions{})
+				Expect(err).To(
+					BeNil(),
+					"while deleting pod {%s} in namespace {%s}",
+					podName,
+					namespaceObj.Name,
+				)
+
+				By("verifying pod count as 0")
+				podCount := ops.GetPodCountEventually(namespaceObj.Name, label, nil, 0)
+				Expect(podCount).To(Equal(0), "while verifying pod count")
+
+				By("deleting the PVC with StorageClass " + scName)
+				pvName := ops.GetPVNameFromPVCName(pvcName)
+				Expect(pvName).To(
+					BeEmpty(),
+					"while getting Spec.VolumeName from "+
+						"PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("deleting above PVC")
+				err = ops.PVCClient.Delete(context.TODO(), pvcName, &metav1.DeleteOptions{})
+				Expect(err).To(
+					BeNil(),
+					"while deleting pvc {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("verifying PVC is deleted")
+				status := ops.IsPVCDeletedEventually(pvcName, namespaceObj.Name)
+				Expect(status).To(
+					BeTrue(),
+					"when checking status of deleted PVC {%s}",
+					pvcName,
+				)
+
+				By("deleting the storageClass " + scName)
```

Comment:
StorageClass deletion is handled as a part of the AfterSuite cleanup, explicit cleanup is not required.
Check suite_test.go file for more context.

Reply:
Those will be deleted after en entire suite completes. I want different storage classes within the suite.

Reply:
This is fine. You can have multiple storageClasses created at the same time. Because you are using GenerateName, they will all be named uniquely.